napari/_qt/_qapp_model/qactions/_view.py

```diff
@@ -74,6 +74,19 @@ def _zoom_out(viewer: Viewer):
     viewer.camera.zoom /= 1.5
 
 
+def _toggle_canvas_ndim(viewer: Viewer):
```

Comment:
```suggestion
def _toggle_canvas_ndim(viewer: ViewerModel):
```

Maybe we should describe it as ViewerModel as it does not touch QT code so it could be reused with an alternative backend. Also, the potential test could use ViewerModel instead of Viewer that is much faster.

Reply:
Done in c1ef7ce and b320523